cmd/xl-storage.go

```diff
@@ -710,7 +710,7 @@ func (s *xlStorage) DeleteVol(volume string, forceDelete bool) (err error) {
 		switch {
 		case os.IsNotExist(err):
 			return errVolumeNotFound
-		case isSysErrNotEmpty(err):
+		case isSysErrNotEmpty(err) || os.IsExist(err):
```

Comment:
Move this to check inside 
isSysErrNotEmpty() and os.IsExist should be only present for runtime.GOOS == "solaris" 

Because on Linux it gives a completely different meaning, this also would avoid changes in entire xl-storage.go

Reply:
Good call, I'll make that change.

Reply:
OK, I've made these updates, re-squashed and force pushed back to the branch (hopefully that's the suggested mechanism here).